File: 002/002.cpp
#include <iostream>
using namespace std;

int total = 0;
int a = 1;
int b = 2;
int c;

int main(){
  while (b<4000000) {
    if (b%2==0) {
      total += b;
    }
    c = b;
    b = a + b;
    a = c;
  }
  cout << total << "\n";
  return 0;
}

File: 002/002.rs
fn main(){
    let mut a = 1;
    let mut b = 2;
    let mut total = 0;

    while b < 4000000{
        if b % 2 == 0{
            total += b;
        }
        (a, b) = (b, a+b)
    }
    println!("Total is {}.", total)
}